superset-frontend/src/views/CRUD/data/dataset/AddDataset/types.tsx

```diff
@@ -39,12 +37,16 @@ interface DatasetReducerPayloadType {
   value?: string;
 }
 
+export type Schema = {
+  schema?: string | null | undefined;
+};
+
 export type DSReducerActionType =
```

Comment:
```suggestion
export type DSReducerActionType =
  | {
      type: DatasetActionType.selectDatabase | DatasetActionType.selectTable;
      payload: Partial<DatasetObject>;
    }
  | {
      type: DatasetActionType.changeDataset | DatasetActionType.selectSchema;
      payload: DatasetReducerPayloadType;
    };
```
(3/3): If you change the `DSReducerActionType` to look like this, the other 2 parts of this state/type change should work and schema will be properly described as a string without needing `as string`